internal/service/datazone/domain.go

```diff
@@ -0,0 +1,510 @@
+	if out, ok := outputRaw.(*datazone.GetDomainOutput); ok {
+		return out, err
+	}
+
+	return nil, err
+}
+
+func waitDomainDeleted(ctx context.Context, conn *datazone.Client, id string, timeout time.Duration) (*datazone.GetDomainOutput, error) {
+	stateConf := &retry.StateChangeConf{
+		Pending: enum.Slice(awstypes.DomainStatusAvailable, awstypes.DomainStatusDeleting),
+		Target:  []string{},
+		Refresh: statusDomain(ctx, conn, id),
+		Timeout: timeout,
+	}
+
+	outputRaw, err := stateConf.WaitForStateContext(ctx)
+	if out, ok := outputRaw.(*datazone.GetDomainOutput); ok {
+		return out, err
+	}
+
+	return nil, err
+}
+
+func statusDomain(ctx context.Context, conn *datazone.Client, id string) retry.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		out, err := findDomainByID(ctx, conn, id)
+		if tfresource.NotFound(err) {
+			return nil, "", nil
+		}
+
+		if err != nil {
+			return nil, "", err
+		}
+
+		return out, string(out.Status), nil
+	}
+}
+
+func findDomainByID(ctx context.Context, conn *datazone.Client, id string) (*datazone.GetDomainOutput, error) {
+	in := &datazone.GetDomainInput{
+		Identifier: aws.String(id),
+	}
+
+	out, err := conn.GetDomain(ctx, in)
+	if err != nil {
+		if isResourceMissing(err) {
+			return nil, &retry.NotFoundError{
+				LastError:   err,
+				LastRequest: in,
+			}
+		}
+
+		return nil, err
+	}
+
+	if out == nil {
+		return nil, tfresource.NewEmptyResultError(in)
+	}
+
+	return out, nil
+}
+
+func isResourceMissing(err error) bool {
+	// DataZone returns a 403 when the domain does not exist
+	// AccessDeniedException: User is not permitted to perform operation: GetDomain
+	return errs.IsA[*awstypes.ResourceNotFoundException](err) || errs.IsAErrorMessageContains[*awstypes.AccessDeniedException](err, "is not permitted to perform")
+}
+
+func flattenSingleSignOn(ctx context.Context, apiObject *awstypes.SingleSignOn) (types.List, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	elemType := types.ObjectType{AttrTypes: singleSignOnAttrTypes}
+
+	if apiObject == nil {
+		return types.ListNull(elemType), diags
+	}
+
+	obj := map[string]attr.Value{
+		"type":            flex.StringValueToFramework(ctx, apiObject.Type),
+		"user_assignment": flex.StringValueToFramework(ctx, apiObject.UserAssignment),
+	}
+	objVal, d := types.ObjectValue(singleSignOnAttrTypes, obj)
+	diags.Append(d...)
+
+	listVal, d := types.ListValue(elemType, []attr.Value{objVal})
+	diags.Append(d...)
+
+	return listVal, diags
+}
+
+func expandSingleSignOn(tfList []singleSignOnData) *awstypes.SingleSignOn {
+	if len(tfList) == 0 {
+		return nil
+	}
+
+	tfObj := tfList[0]
+	apiObject := &awstypes.SingleSignOn{}
+
+	if !tfObj.Type.IsNull() {
+		apiObject.Type = awstypes.AuthType(tfObj.Type.ValueString())
+	}
+
+	if !tfObj.UserAssignment.IsNull() {
+		apiObject.UserAssignment = awstypes.UserAssignment(tfObj.UserAssignment.ValueString())
+	}
+
+	return apiObject
+}
+
+type resourceDomainData struct {
```

Comment:
Naming convention is normally:

<resourcename><type>Model.

example: [customModelResourceModel](https://github.com/hashicorp/terraform-provider-aws/blob/e6da217565544de7ce3a1e0e51e397e338145ee8/internal/service/bedrock/custom_model.go#L619)

```suggestion
type domainResourceModel struct {
```

Reply:
Ah awesome, updated! :)